iothub_client/src/iothub_message.c

```diff
@@ -43,8 +43,8 @@ static bool ContainsOnlyUsAscii(const char* asciiValue)
     const char* iterator = asciiValue;
     while (iterator != NULL && *iterator != '\0')
     {
-        // Allow only printable ascii char
-        if (*iterator < ' ' || *iterator > '~')
+        // Allow only printable ascii char (excluding space ' ').
```

Comment:
This is still not right. These chars are NOT allowed
" ( ) , / : ; < = > ? @ { }

Reply:
I love how off the original code was from the specs lol.

Reply:
Due to validation results, we should leave this as is.  Some protocols allows all characters [dec 32, dec 126], while others are limited.  It is also c2d vs d2c dependent.  I do not know of a way the code can determine which protocol is being used at this level, and I do not think it should be responsible to.  This puts the onus on the IoT Hub documentation to inform the user, and if the user uses invalid ascii characters, they will encounter an error elsewhere.